### DISCUSSION TITLE
Guards inside onDone

### DISCUSSION THREAD

**Discussion by GabrielAPineiro** [statelyai/xstate#4978] [Q&A]
I've noticed that when creating a guard inside setup and then using it inside an event on the "on" part of the state, I have access to context, which is nice;
But then, I tried to use the same guard inside an onDone I get an error saying the guard wasn't implemented. And If I try to use an inline guard, I only have access to event and context but is the context defined inside createMachine, which I'm currently using the "context: ({ input }) =>" type of definition. Why is that? Am I doing something wrong?

**Comment by davidkpiano** [comment]
Can you please share the code that is causing the problem?
